### PR TITLE
rio: 0.1.0 -> 0.1.1 + mark as broken on darwin

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/raphamorim/rio/blob/v0.1.1/CHANGELOG.md

cc @TornaxO7 @otavio @oluceps 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TornaxO7** [comment]
There's already https://github.com/NixOS/nixpkgs/pull/325243

**Comment by GaetanLepage** [comment]
> There's already #325243

Oh indeed.
In this one I added the `testVersion` test. Apart from that, it is the same as the bot PR.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>rio</li>
    <li>rio.terminfo</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325469` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>rio</li>
    <li>rio.terminfo</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325469` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>rio</li>
    <li>rio.terminfo</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Fails on `darwin` with:
```
error: linking with `/nix/store/k8mrxviw965lv59hqbln3297jwfffm2w-clang-wrapper-16.0.6/bin/cc` failed: exit status: 1
  |
  = note: env -u IPHONEOS_DEPLOYMENT_TARGET -u TVOS_DEPLOYMENT_TARGET LC_ALL="C" PATH="/nix/store/8wjjv08wbm62l74xbn6nh3w48xv44hfl-rustc-1.78.0/lib/rustlib/aarch64-apple-darwin/bin:>
  = note: ld: framework not found MetalKit
          clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: could not compile `wa` (lib test) due to 1 previous error
```

**Comment by GaetanLepage** [comment]
> Fails on `darwin` with:

It is already the case for the currently packaged version (0.1.0), so let's mark it as broken.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325469` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>rio</li>
    <li>rio.terminfo</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/terminal-emulators/rio/default.nix [hunk 1]
```diff
@@ -26,6 +26,9 @@
 
 , withWayland ? !stdenv.isDarwin
 , wayland
+
+, testers
+, rio
 }:
 let
   rlinkLibs = if stdenv.isDarwin then [
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/rio/default.nix:34]: We probably just need to add MetalKit here

**Review by SuperSandro2000 — COMMENTED** [review]
Also the version update commit has the wrong starting version, that should be 0.1.0

File: pkgs/applications/terminal-emulators/rio/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/terminal-emulators/rio/default.nix:34]: That was it indeed.
However we have a test failing:

```
failures:

---- corcovado/src/sys/unix/eventedfd.rs - sys::unix::eventedfd::EventedFd (line 31) stdout ----
Test executable failed (exit status: 101).

stderr:
thread 'main' panicked at corcovado/src/sys/unix/eventedfd.rs:24:16:
called `Result::unwrap()` on an `Err` value: Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" }
stack backtrace:
   0:        0x104d4b838 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h461bbd67a6621422
   1:        0x104ce5828 - core::fmt::write::heed46d935b88c574
   2:        0x104d4e6a0 - std::io::Write::write_fmt::h059fcaa37c7fb6b7
   3:        0x104d4b5d8 - std::sys_common::backtrace::print::he45faad0cc62205b
   4:        0x104d4c700 - std::panicking::default_hook::{{closure}}::h079849c23e14ba8c
   5:        0x104d4c268 - std::panicking::default_hook::h23406b9c4937065d
   6:        0x104d4d330 - std::panicking::rust_panic_with_hook::h590310b8f01355ea
   7:        0x104d4ce8c - std::panicking::begin_panic_handler::{{closure}}::h1e72464f58135f03
   8:        0x104d4bb38 - std::sys_common::backtrace::__rust_end_short_backtrace::h569fec281ca4c9f5
   9:        0x104d4cae8 - _rust_begin_unwind
  10:        0x104d52534 - core::panicking::panic_fmt::h78e83b1750882f24
  11:        0x104d529c4 - core::result::unwrap_failed::h4109b0718192d6d7
  12:        0x104cf5710 - rust_out::main::h4fb46e86f974d2f5
  13:        0x104cf3c34 - core::ops::function::FnOnce::call_once::hf02b6f540bebbb60
  14:        0x104cf2fec - std::sys_common::backtrace::__rust_begin_short_backtrace::ha5dea535a3027ddc
  15:        0x104cf3690 - std::rt::lang_start::{{closure}}::hcd09604f56e8968b
  16:        0x104d4c7c0 - std::panicking::try::h306a92dbe5fcd94d
  17:        0x104d42bd4 - std::rt::lang_start_internal::hcd3b15c34f03da8c
  18:        0x104cf3668 - std::rt::lang_start::h809b4565db301f43
  19:        0x104cf576c - _main



failures:
    corcovado/src/sys/unix/eventedfd.rs - sys::unix::eventedfd::EventedFd (line 31)

test result: FAILED. 55 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 12.66s
```

Comment (GaetanLepage) [reply pkgs/applications/terminal-emulators/rio/default.nix:34]: Maybe it tries to access the GUI ?

**Comment by TornaxO7** [comment]
Should I approve? For me, it works on x86_64-linux

File: pkgs/applications/terminal-emulators/rio/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/terminal-emulators/rio/default.nix:34]: I don't know, just leave it marked broken

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**